### PR TITLE
Feature Toggles: Switch feature toggle admin page over to k8s API

### PR DESCRIPTION
**Pull request by mmandrus**
**What is this feature?**

Updates the feature toggle admin page to integrate with the new k8s style apis instead of the legacy ones. Also implements the webhook in the new API handler.

Verified locally and in the ephemeral instance below. Removed the /api toggles entirely as the entire feature is still Experimental.

### CODE REVIEW

File: pkg/apis/featuretoggle/v0alpha1/types.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ type ToggleStatus struct {
 	Enabled bool `json:"enabled"`
 
 	// Can this flag be updated
-	Writeable bool `json:"writeable,omitempty"`
+	Writeable bool `json:"writeable"`
```
Comment (mmandrus) [inline pkg/apis/featuretoggle/v0alpha1/types.go:102]: just wanted to include the false values explicitly

File: public/app/features/admin/AdminFeatureTogglesAPI.ts [hunk 2]
```diff
@@ -26,37 +60,86 @@ const backendSrvBaseQuery =
...
-    getFeatureToggles: builder.query<FeatureToggle[], void>({
-      query: () => ({ url: '/featuremgmt' }),
-    }),
-    updateFeatureToggles: builder.mutation<void, FeatureToggle[]>({
-      query: (updatedToggles) => ({
-        url: '/featuremgmt',
-        method: 'POST',
-        body: { featureToggles: updatedToggles },
+const getK8sAPI = () => {
+  return createApi({
```
Comment (mmandrus) [inline public/app/features/admin/AdminFeatureTogglesAPI.ts:64]: only real difference here is the urls and the conversion of the body in the patch method

File: public/app/features/admin/AdminFeatureTogglesAPI.ts [hunk 3]
```diff
@@ -26,37 +60,86 @@ const backendSrvBaseQuery =
...
+  toggles.forEach((t) => {
+    patchBody.enabled[t.name] = t.enabled;
+  });
+  return patchBody;
+}
+
+const togglesApi = config.featureToggles.kubernetesFeatureToggles ? getK8sAPI() : getLegacyAPI();
+let { useGetManagerStateQuery, useGetFeatureTogglesQuery, useUpdateFeatureTogglesMutation } = togglesApi;
+
+if (config.featureToggles.kubernetesFeatureToggles) {
```
Comment (mmandrus) [inline public/app/features/admin/AdminFeatureTogglesAPI.ts:121]: Using this function wrapping approach, it's transparent to any code calling these functions. However, not sure it is worth the trouble. We could just not export `useGetFeatureTogglesQuery` at all, and instead export an explicit wrapper function.

File: public/app/features/admin/AdminFeatureTogglesPage.tsx [hunk 4]
```diff
@@ -59,7 +59,7 @@ export default function AdminFeatureTogglesPage() {
           {isError && getErrorMessage()}
           {isLoading && 'Fetching feature toggles'}
           {featureMgmtState?.allowEditing && <EditingAlert />}
-          {featureToggles && (
+          {!isLoading && featureToggles && (
```
Comment (mmandrus) [inline public/app/features/admin/AdminFeatureTogglesPage.tsx:62]: for some reason, this was breaking things with the new k8s api. this should make react more predictable

**Review by ryantxu — APPROVED** [review]
Looks great!  next step is to run it in production for a bit... then remove the flag + /api endpoint 🌮 🎉

File: pkg/services/featuremgmt/models.go [hunk 5]
```diff
@@ -151,3 +151,8 @@ type FeatureManagerState struct {
 	RestartRequired bool `json:"restartRequired"`
 	AllowEditing    bool `json:"allowEditing"`
 }
+
+type FeatureToggleWebhookPayload struct {
+	FeatureToggles map[string]string `json:"feature_toggles"`
+	User           string            `json:"user"`
```
Comment (ryantxu) [inline pkg/services/featuremgmt/models.go:157]: lets revist this when we move everything to the /apis/ package

**Comment by torkelo** [comment]
cool!

**Comment by mmandrus** [comment]
LGTM and worked on the ephemeral instance. Nice work 👍

**Pull request merged**